### PR TITLE
Add auth to get endpoints

### PR DESCRIPTION
**Pull request by samucorea**
# Description

Add auth to get endpoints, so not anyone can get a patient by id

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

### CODE REVIEW

File: NoNicotineAPI/Controllers/LoginController.cs [hunk 1]
```diff
@@ -29,11 +29,6 @@ public async Task<IActionResult> Post(AuthenticateQuery request)
         {
             var result = await _mediator.Send(request);
 
-            if (result.Data == null)
```
Comment (gasimon12) [inline NoNicotineAPI/Controllers/LoginController.cs:32]: Why did you delete this :(

File: NoNicotineAPI/Controllers/TherapistsController.cs [hunk 2]
```diff
@@ -30,12 +33,19 @@ public async Task<IActionResult> CreateTherapist(CreateTherapistCommand request)
         }
 
         [HttpGet]
-        [Route("{id}")]
-        public async Task<IActionResult> GetTherapist(string id)
+        [Authorize(Roles = "therapist")]
+        public async Task<IActionResult> GetTherapist()
         {
+            if (HttpContext.User.Identity is not ClaimsIdentity identity)
```
Comment (gasimon12) [inline NoNicotineAPI/Controllers/TherapistsController.cs:39]: You have a different implementation here than the one you used for patient.

File: NoNicotine_Business/Handler/CreatePatientCommandHandler.cs [hunk 3]
```diff
@@ -23,14 +23,16 @@ public class CreatePatientCommandHandler : IRequestHandler<CreatePatientCommand,
...
-        public CreatePatientCommandHandler( UserManager<IdentityUser> userManager, 
-            ILogger<CreatePatientCommandHandler> logger, AppDbContext dbContext)
+        public CreatePatientCommandHandler( UserManager<IdentityUser> userManager,
+            ILogger<CreatePatientCommandHandler> logger, IPatientRepository patientRepository, AppDbContext context)
         {
             _userManager = userManager;
             _logger = logger;
-            _context = dbContext;
+            _patientRepository = patientRepository;
+            _context = context;
```
Comment (gasimon12) [inline NoNicotine_Business/Handler/CreatePatientCommandHandler.cs:35]: Why do you have both context and repository here, but in other classes only have repository (f.e. GetPatientQueryHandler)??

File: NoNicotineAPI/Controllers/TherapistsController.cs (hunk 2 above)
Comment (samucorea) [reply NoNicotineAPI/Controllers/TherapistsController.cs:39]: made it the same. fixed

**Review by gasimon12 — APPROVED** [review]
